src/server/cards/base/library.js

```diff
@@ -1,11 +1,18 @@
 import Card from 'cards/Card';
-import { Set } from 'immutable';
 
 export default class Library extends Card {
-  static name = 'Library';
   static cost = 5;
   static types = new Set(['Action']);
   async onPlay(player) {
-    player.draw(7 - player.hand.size);
+    while ((player.hand.size < 7) && (player.deck.size + player.discardPile.size > 0)) {
+    	await player.draw(1, async card => {
+    		if (card.types.has('Action')) {
+    			const choice = await player.selectOption(['Put ' + card.title + ' in hand', 'Set ' + card.title + ' aside']);
+    			return [player.hand, player.aside][choice];
+    		}
+    		return player.hand;
+  		});
+    }
+    await player.aside.forEach(async card => await player.discard(card));
```

Comment:
same thing with `forEach`
do we really need `aside` to be part of `Player`? maybe it would make more sense to just make a `Pile` in this function and hold the cards there